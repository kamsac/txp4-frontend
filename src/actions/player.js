import dotenvConfiguration from '../dotenv-configuration';
import PlayerResourceMock from '../resources/player/mock';
import PlayerResource from '../resources/player';

export const RECEIVE_PLAYER = 'player:receivePlayer';
export function receivePlayer(player) {
  return {
    type: RECEIVE_PLAYER,
    payload: { player },
  };
}

function fetchPlayer() {
  const resource = dotenvConfiguration.API_URL ? PlayerResource : PlayerResourceMock;

  return (dispatch) => {
    resource.getPlayer()
      .then((response) => {
        dispatch(receivePlayer(response.data));
      });
  };
}

export const REQUEST_PLAYER = 'player:requestPlayer';
export function requestPlayer() {
  return (dispatch) => {
    dispatch(fetchPlayer());
  };
}
