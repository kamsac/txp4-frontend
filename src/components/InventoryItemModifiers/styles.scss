$positive-color: #080;
$negative-color: #f00;

.InventoryItemModifiers {
  padding: 0;
}

.InventoryItemModifiers-modifier {
  list-style-type: none;
  line-height: 2rem;
}

.InventoryItemModifiers-modifier--positive {
  color: $positive-color;
}

.InventoryItemModifiers-modifier--negative {
  color: $negative-color;
}
