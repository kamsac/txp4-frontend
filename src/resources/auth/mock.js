export const ACCESS_TOKEN = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiaWF0IjoxNTE2MjM5MDIyLCJsb2dpbiI6Imx5bWFrIiwibmljayI6IiRzJG8kYTYwbCRjODJ5JGZhM20kYzgyYSRhNjBrIn0.aaKzXEZP2tovhBAqPJYBdmbLzKjfdBsuX3FaB58JZvE';

const AuthResourceMock = {
  login() {
    return new Promise((resolve) => {
      const response = {
        data: {
          token: ACCESS_TOKEN,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default AuthResourceMock;
