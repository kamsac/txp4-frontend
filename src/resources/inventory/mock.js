const ITEMS = [
  {
    id: "a1a1a1",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: true,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "b2b2b2",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "c3c3c3",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "d4d4d4",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "e5e5e5",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "f6f6f6",
    name: "Duper engine",
    vendorId: 'aaa',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1.5,
      },
      {
        name: "acceleration",
        value: 3,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "f7f7f7",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "g8g8g8",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "h9h9h9",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "i0i0i0",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "j1j1j1",
    name: "Transmission",
    vendorId: 'ccc',
    equipRegion: "transmission",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.7,
      },
      {
        name: "acceleration",
        value: 1.5,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
];

const InventoryResourceMock = {
  getItems() {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  equipItem(item) {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(ITEMS), 100);
    });
  },
};

export default InventoryResourceMock;
