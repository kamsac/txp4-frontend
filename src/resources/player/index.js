import axios from 'axios';
import dotenvConfiguration from '../../dotenv-configuration';

const PlayerResource = {
  getPlayer(playerLogin) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}`);
  },

  getInventory(playerLogin) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory`);
  },

  equipItem(playerLogin, itemId) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory/${itemId}/equip`);
  },

  unequipItem(playerLogin, itemId) {
    return axios.get(`${dotenvConfiguration.API_URL}/players/${playerLogin}/inventory/${itemId}/unequip`);
  },
};

export default PlayerResource;
