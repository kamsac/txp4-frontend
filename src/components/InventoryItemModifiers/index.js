import React from 'react';
import './styles.scss';

export default class InventoryItemModifiers extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const modifiers = this.props.item.modifiers
      .filter((modifier) => modifier.value !== 1)
      .map((modifier) => {
        const percent = Math.round(modifier.value * 100) + '%';
        return (
          <li key={modifier.name}>{`${percent} of ${modifier.name}`}</li>
        );
      });

    return (
      <ul className="InventoryItemModifiers">
        {modifiers}
      </ul>
    )
  }
}
