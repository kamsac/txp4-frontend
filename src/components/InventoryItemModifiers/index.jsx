import React from 'react';
import classNames from 'classnames';
import './styles.scss';
import { ItemPropTypesShape } from '../../prop-types';

export default class InventoryItemModifiers extends React.Component {
  render() {
    const modifiers = this.props.item.modifiers
      .filter(modifier => modifier.value !== 1)
      .sort((modifierA, modifierB) => modifierB.value - modifierA.value)
      .map((modifier) => {
        const percent = `${Math.round(modifier.value * 100)}%`;
        const className = classNames('InventoryItemModifiers-modifier', {
          'InventoryItemModifiers-modifier--positive': modifier.value > 1,
          'InventoryItemModifiers-modifier--negative': modifier.value < 1,
        });
        return (
          <li key={modifier.name} className={className}>{`${percent} of ${modifier.name}`}</li>
        );
      });

    return (
      <ul className="InventoryItemModifiers">
        {modifiers}
      </ul>
    );
  }
}

InventoryItemModifiers.propTypes = {
  item: ItemPropTypesShape.isRequired,
};
