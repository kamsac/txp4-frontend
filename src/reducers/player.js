import inventoryReducer from './inventory';
import { RECEIVE_PLAYER } from '../actions/player';
import { EQUIP_ITEM, RECEIVE_INVENTORY, REQUEST_INVENTORY } from '../actions/inventory';

const initialState = {
  login: '',
  nick: '',
  inventory: {
    items: [],
  },
};

export default function playerReducer(state = initialState, action) {
  switch (action.type) {
    case RECEIVE_PLAYER:
      return Object.assign({}, state, action.payload.player);

    case REQUEST_INVENTORY:
    case RECEIVE_INVENTORY:
    case EQUIP_ITEM: {
      const inventory = inventoryReducer(state.inventory, action);
      return Object.assign({}, state, { inventory });
    }
    default:
      return state;
  }
}
