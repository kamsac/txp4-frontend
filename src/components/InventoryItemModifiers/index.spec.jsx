import React from 'react';
import { shallow } from 'enzyme';
import InventoryItemModifiers from './index';
import { ITEMS } from '../../resources/player/mock';

test('<InventoryItemModifiers>', () => {
  const component = shallow(<InventoryItemModifiers item={ITEMS[0]} />);
  expect(component).toMatchSnapshot();
});
