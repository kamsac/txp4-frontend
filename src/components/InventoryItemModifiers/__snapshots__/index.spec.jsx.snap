// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InventoryItemModifiers> 1`] = `
<ul
  className="InventoryItemModifiers"
>
  <li
    className="InventoryItemModifiers-modifier InventoryItemModifiers-modifier--positive"
    key="grip"
  >
    150% of grip
  </li>
  <li
    className="InventoryItemModifiers-modifier InventoryItemModifiers-modifier--positive"
    key="acceleration"
  >
    120% of acceleration
  </li>
  <li
    className="InventoryItemModifiers-modifier InventoryItemModifiers-modifier--positive"
    key="breaking"
  >
    120% of breaking
  </li>
  <li
    className="InventoryItemModifiers-modifier InventoryItemModifiers-modifier--positive"
    key="steering"
  >
    110% of steering
  </li>
</ul>
`;
